src/deluge/deluge.cpp

```diff
@@ -451,17 +451,19 @@ void setupBlankSong() {
 
 /// Can only happen after settings, which includes default settings, have been read
 void setupStartupSong() {
-	auto templatePath = "SONGS/DEFAULT.XML";
 	auto startupSongMode = FlashStorage::defaultStartupSongMode;
 
-	if (startupSongMode == StartupSongMode::BLANK) {
-		setupBlankSong();
-	}
-	else if (startupSongMode == StartupSongMode::TEMPLATE && !storageManager.fileExists(templatePath)) {
-		setupBlankSong();
-		currentSong->writeTemplateSong(templatePath);
-	}
-	else {
+	switch (startupSongMode) {
+	case StartupSongMode::TEMPLATE:
+	case StartupSongMode::LASTOPENED:
+	case StartupSongMode::LASTSAVED: {
+		if (startupSongMode == StartupSongMode::TEMPLATE) {
+			auto templatePath = "SONGS/DEFAULT.XML";
+			setupBlankSong();
+			if (!storageManager.fileExists(templatePath)) {
+				currentSong->writeTemplateSong(templatePath);
+			}
```

Comment:
This code can go under the `case StartupSongMode::TEMPLATE:`

Reply:
hum no, because there's also the logic to load the DEFAULT.XML song if it does already exist

Reply:
Right, but it can fall through that case and then execute the rest of the logic.

Reply:
ok didn't know this trick, let me change this when i'm back